internal/provider/webhook_recipient_resource.go

```diff
@@ -247,3 +290,139 @@ func (r *webhookRecipientResource) Delete(ctx context.Context, req resource.Dele
 		}
 	}
 }
+
+func webhookTemplateSchema() schema.SetNestedBlock {
+	return schema.SetNestedBlock{
+		Description: "Template for custom webhook payloads",
+		NestedObject: schema.NestedBlockObject{
+			Attributes: map[string]schema.Attribute{
+				"type": schema.StringAttribute{
+					Required:    true,
+					Description: "The type of the webhook template",
+					Validators: []validator.String{
+						stringvalidator.OneOf(webhookTemplateTypes...),
+					},
+				},
+				"body": schema.StringAttribute{
```

Comment:
probably should do a length validator on this but I didn't find a validation in the manager to base it on so was going to hold off as a follow-up and make it match in both places